### ISSUE TITLE
Change ASP.NET Core 2.0 templates to use new static WebHost API

### ISSUE THREAD

**Issue by DamianEdwards** [dotnet/templating#664]
All the ASP.NET Core 2.0 templates should have their `Program.Main` method updated to use the new static `WebHost.CreateDefaultBuilder()` API, e.g.:
``` c#
public static Main(string[] args)
{
    var host = WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .Build();
}
```

I think this will also require all the templates to ensure they have the `<UserSecretsId>` MSBUILD property set in their CSPROJ files too.

**Comment by DamianEdwards** [comment]
@glennc @JunTaoLuo @danroth27 @mlorbetske

**Comment by mlorbetske** [comment]
Fixed by #671
